src/features/search/components/__tests__/SearchResults.native.test.tsx

```diff
@@ -170,11 +170,57 @@ describe('SearchResults component', () => {
     beforeAll(() => {
       mockSettings.mockReturnValue({ data: { appEnableCategoryFilterPage: true } })
     })
+
+    it('should display type filter button', () => {
+      const { queryByTestId } = render(<SearchResults />)
+
+      expect(queryByTestId('typeButton')).toBeTruthy()
+    })
+
+    it('should redirect to the general filters page when clicking on the type filter button', () => {
+      const { getByTestId } = render(<SearchResults />)
+      const typeButton = getByTestId('typeButton')
+
+      fireEvent.press(typeButton)
+
+      expect(navigate).toHaveBeenNthCalledWith(1, 'SearchFilter')
+    })
+
+    it.each`
+      type               | params
+      ${'duo offer'}     | ${{ offerIsDuo: true }}
+      ${'digital offer'} | ${{ offerTypes: { isDigital: true, isEvent: false, isThing: false } }}
+      ${'event offer'}   | ${{ offerTypes: { isDigital: false, isEvent: true, isThing: false } }}
+      ${'thing offer'}   | ${{ offerTypes: { isDigital: false, isEvent: false, isThing: true } }}
+    `(
+      'should display an icon and change color in type button when $type selected',
```

Comment:
`should display an icon and change color in type button when has $type selected`

(ajout de has)